designcompose/src/main/java/com/android/designcompose/VariableManager.kt

```diff
@@ -170,20 +171,21 @@ internal object VariableManager {
 
     internal fun init(docId: DesignDocId, map: VariableMap) {
 
-        // Remove old entries for docId
-        val oldVarMap = docVarMap[docId.id]
-        oldVarMap?.collectionsMap?.forEach { varMap.collectionsMap.remove(it.key) }
-        oldVarMap?.collectionNameMapMap?.forEach { varMap.collectionNameMapMap.remove(it.key) }
-        oldVarMap?.variablesMap?.forEach { varMap.variablesMap.remove(it.key) }
-        oldVarMap?.variableNameMapMap?.forEach { varMap.variableNameMapMap.remove(it.key) }
-
-        // Add new entries for docId
-        docVarMap[docId.id] = map
-        varMap.collectionsMap.putAll(map.collectionsMap)
-        varMap.collectionNameMapMap.putAll(map.collectionNameMapMap)
-        varMap.variablesMap.putAll(map.variablesMap)
-        varMap.variableNameMapMap.putAll(map.variableNameMapMap)
-
+        varMap = varMap.copy {
+            // Remove old entries for docId
+            val oldVarMap = docVarMap[docId.id]
+            oldVarMap?.collectionsMap?.forEach { collections.remove(it.key) }
+            oldVarMap?.collectionNameMapMap?.forEach { collectionNameMap.remove(it.key) }
+            oldVarMap?.variablesMap?.forEach { variables.remove(it.key) }
+            oldVarMap?.variableNameMapMap?.forEach { variableNameMap.remove(it.key) }
```

Comment:
This chunk doesn't need to be in the copy block does it?

Reply:
It does, it's modifying the varMap. I'll re-add the this's to make it more clear